src/scene/mesh-instance.js

```diff
@@ -455,6 +459,29 @@ class MeshInstance {
         return bindGroup;
     }
 
+    /**
+     * Performs a raycast with a given ray against the mesh
+     *
+     * @param {Ray} ray - the ray to perform the intersect test with
+     * @returns {Vec3} Null if there is no intersection or the point of intersection
+     * in world space
+     */
+    rayCast(ray) {
+        _worldTransformInverted.copy(this.node.getWorldTransform());
+        _worldTransformInverted.invert();
+        _transformedRay.set(ray.origin, ray.direction);
```

Comment:
This is a crutch for a missing `Ray#copy`, maybe just add that?

Reply:
`Ray#copy` is in the engine now... just wondering, is this still planned/worked-on?

Reply:
it's not being worked on at the moment unfortunately, but hopefully can be picked up a finished one day.

Reply:
Why does the meshInstance.raycast return the intersection point and a mesh.raycast the distance? Preferably you would get either in both cases. (btw awesome work on this!)